gradio/components.py

```diff
@@ -2236,9 +2236,14 @@ def save_flagged(self, dir, label, data, encryption_key):
         """
         Returns: (str) path to file
         """
-        return self.save_flagged_file(
-            dir, label, None if data is None else data[0]["data"], encryption_key
-        )
+        if isinstance(data, dict):
+            return self.save_flagged_file(
+                dir, label, data["data"], encryption_key, data["name"]
+            )
+        else:
+            return self.save_flagged_file(
+                dir, label, None if data is None else data[0]["data"], encryption_key
+            )
 
```

Comment:
@abidlabs Do only the file tests `test_as_component_as_output` pass in a data array instead of a dict here? If so I can update the tests and remove this check.

Reply:
I think so yeah. I would test with single files and multiple files and if everything works, go ahead and update the tests and remove the check